src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,507 @@
+     */
+    public void writeToNBT(NBTTagCompound compound)
+    {
+        BlockPosUtil.writeToNBT(compound, TAG_LOCATION, this.location);
+        compound.setBoolean(TAG_TAKEN, taken);
+        compound.setInteger(TAG_STAGE, fieldStage.ordinal());
+        compound.setInteger(TAG_LENGTH_PLUS, lengthPlusX);
+        compound.setInteger(TAG_WIDTH_PLUS, widthPlusZ);
+        compound.setInteger(TAG_LENGTH_MINUS, lengthMinusX);
+        compound.setInteger(TAG_WIDTH_MINUS, widthMinusZ);
+        inventory.writeToNBT(compound);
+    }
+
+    /**
+     * Save data to NBT compound.
+     * Writes the {@link #location} value.
+     *
+     * @param compound      {@link net.minecraft.nbt.NBTTagCompound} to write data to.
+     */
+    public void readFromNBT(NBTTagCompound compound)
+    {
+        location = BlockPosUtil.readFromNBT(compound, TAG_LOCATION);
+        taken = compound.getBoolean(TAG_TAKEN);
+        fieldStage = FieldStage.values()[compound.getInteger(TAG_STAGE)];
+        lengthPlusX = compound.getInteger(TAG_LENGTH_PLUS);
+        widthPlusZ = compound.getInteger(TAG_WIDTH_PLUS);
+        lengthMinusX = compound.getInteger(TAG_LENGTH_MINUS);
+        widthMinusZ = compound.getInteger(TAG_WIDTH_MINUS);
+        inventory = new InventoryField("Scarecrow", true);
+        inventory.readFromNBT(compound);
+    }
+
+    /**
+     * Has the field been taken?
+     * @return true if the field is not free to use, false after releasing it.
+     */
+    public boolean isTaken() {
+        return this.taken;
+    }
+
+    /**
+     * Sets the field taken.
+     * @param taken is field free or not
+     */
+    public void setTaken(boolean taken)
+    {
+        this.taken = taken;
+    }
+
+    /**
+     * Checks if the field has been planted.
+     * @return true if there are crops planted.
+     */
+    public FieldStage getFieldStage()
+    {
+        return this.fieldStage;
+    }
+
+    /**
+     * Sets if there are any crops planted.
+     * @param fieldStage true after planting, false after harvesting.
+     */
+    public void setFieldStage(FieldStage fieldStage)
+    {
+        this.fieldStage = fieldStage;
+    }
+
+    /**
+     * Checks if the field needs work (planting, hoeing).
+     * @return true if so.
+     */
+    public boolean needsWork()
+    {
+        return this.needsWork;
+    }
+
+    /**
+     * Sets that the field needs work.
+     * @param needsWork true if work needed, false after completing the job.
+     */
+    public void setNeedsWork(boolean needsWork)
+    {
+        this.needsWork = needsWork;
+    }
+
+    /**
+     * Getter for MAX_RANGE.
+     * @return the max range.
+     */
+    private static int getMaxRange()
+    {
+        return MAX_RANGE;
+    }
+
+    /**
+     * Getter of the seed of the field
+     * @return the ItemSeed
+     */
+    public Item getSeed()
+    {
+        if(inventory.getStackInSlot(0) == null || !(inventory.getStackInSlot(0).getItem()  instanceof IPlantable))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) No need to check for null before an instanceof [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASimplifyConditional)